source/extensions/filters/http/ext_proc/ext_proc.h

```diff
@@ -90,8 +90,10 @@ class Filter : public Logger::Loggable<Logger::Id::filter>,
 
 public:
   Filter(const FilterConfigSharedPtr& config, ExternalProcessorClientPtr&& client)
-      : config_(config), client_(std::move(client)), stats_(config->stats()),
-        processing_mode_(config->processingMode()) {}
+      : config_(config), client_(std::move(client)), stats_(config->stats()), decoding_state_(this),
+        encoding_state_(this), processing_mode_(config->processingMode()) {}
+
+  const FilterConfig* config() const { return config_.get(); }
```

Comment:
return by const ref? or can this be nullptr?

Reply:
Yes, this can and should be a const ref.